iam-policy-autopilot-policy-generation/src/extraction/go/disambiguation.rs

```diff
@@ -215,8 +223,13 @@ impl<'a> GoMethodDisambiguator<'a> {
                         }
                     }
 
+                    // Use AST-extracted fields if available (only top-level fields, no nested map keys)
+                    if let Some(fields) = struct_fields {
+                        return Some(fields.clone());
+                    }
+
+                    // Fallback to string parsing for backwards compatibility
                     let value_str = value.as_string();
-                    // Extract field names from struct literals
                     if value_str.contains('{') && value_str.contains('}') {
                         Some(self.extract_struct_field_names(value_str))
```

Comment:
Do we really need this? Can this ever work?

Reply:
We don't, removed.